src/main.cpp

```diff
@@ -23,8 +23,20 @@ constexpr int kFatalErrorOnStartupExitCode = 1;
 constexpr int kParseCmdlineArgsErrorExitCode = 2;
 
 int runMixxx(MixxxApplication* app, const CmdlineArgs& args) {
-    auto coreServices = std::make_shared<mixxx::CoreServices>(args);
-    MixxxMainWindow mainWindow(app, coreServices);
+    const auto pCoreServices = std::make_shared<mixxx::CoreServices>(args);
+    pCoreServices->initializeSettings();
+    pCoreServices->initializeKeyboard();
+
+    MixxxMainWindow mainWindow(app, pCoreServices);
+    app->installEventFilter(&mainWindow);
+
+    QObject::connect(pCoreServices.get(),
+            &mixxx::CoreServices::initializationProgressUpdate,
+            &mainWindow,
+            &MixxxMainWindow::initializationProgressUpdate);
+    pCoreServices->initialize(app);
```

Comment:
Can this be moved to the CoreServices constructor by passing the QApplication* as another argument?

Reply:
I guess we need the main window for monitoring progress during the initialization?

Reply:
> I guess we need the main window for monitoring progress during the initialization?

Indeed. CoreServices are MixxxMainWindow are classes that communicate with each other via signals, but neither does MixxxMainWindow own CoreServices (as it did before) nor vice versa.

Hence, we need to construct both first, then connect the signals, then initialize them.